frontend/src/metabase/lib/data_grid.js

```diff
@@ -18,7 +18,11 @@ export function multiLevelPivot(
   const valuesByKey = {};
 
   // loop over the primary rows to build trees of column/row header data
-  for (const row of data.rows) {
+  for (const row of pivotData[
+    JSON.stringify(
```

Comment:
Not sure I understand what's happening here at a glance.

Reply:
The diff here is because I stopped splitting out the main data set as "data" and left it in one consolidated object with the various subtotals.

I definitely need to write a comment documenting the main data structures at play here.